### PR TITLE
mgr/dashboard: increase timeout for radosgw-admin command

### PR DESCRIPTION
**Pull request by aaSharma14**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by Pegonzal** [comment]
Hey @aaSharma14, does this fix the issues we've been seeing with the RGW timeouts? How can I reproduce it to test it?
And is there any reason to increase it to 60 in particuallary? Seems like a big increase from 10.

Also, this would need a tracker, right?

Thank you for addressing this!

**Comment by nizamial09** [comment]
> And is there any reason to increase it to 60 in particuallary? Seems like a big increase from 10.

yeah, 60 feels like a big leap. If there is a connection timeout, this is going to hang the mgr for 60sec which might be bad. Does 30sec works here?

**Comment by aaSharma14** [comment]
> Hey @aaSharma14, does this fix the issues we've been seeing with the RGW timeouts? How can I reproduce it to test it? And is there any reason to increase it to 60 in particuallary? Seems like a big increase from 10.

While testing this feature, me and @cloudbehl faced a lot of radosgw-admin timeout issues and we thought of increasing the timeout for this command..there we kept it to 60s and the issue got solved..so went with this change. There's no specific use case where you can reproduce this..this happens randomly mostly with the `radosgw-admin period update --commit command`. We are increasing the timeout to 60s just in case of this particular command..open to decrease it as well.

> Also, this would need a tracker, right?

Yeah will create one

> Thank you for addressing this!

**Comment by aaSharma14** [comment]
> > And is there any reason to increase it to 60 in particuallary? Seems like a big increase from 10.
> 
> yeah, 60 feels like a big leap. If there is a connection timeout, this is going to hang the mgr for 60sec which might be bad. Does 30sec works here?

Yeah might work..depends on testing

**Comment by cloudbehl** [comment]
I agree there is not reproducer we have seen till now, I think we can start with increasing it to 30 sec and then maybe see we need to comeup with formula to calculate the time based on how big the cluster is.

 Also this timeout is something which is introduced by UI team and not backend, we can follow a non-timeout/longer timeout approach taken by backend as well for some commands like period commit.

We have not tested this with large cluster I'm curious how it will behave in that case. We maybe need to ask the team to test on bigger cluster as well.

@Pegonzal @aaSharma14 @nizamial09

**Comment by nizamial09** [comment]
> I agree there is not reproducer we have seen till now, I think we can start with increasing it to 30 sec and then maybe see we need to comeup with formula to calculate the time based on how big the cluster is.
> 
> Also this timeout is something which is introduced by UI team and not backend, we can follow a non-timeout/longer timeout approach taken by backend as well for some commands like period commit.
> 
> We have not tested this with large cluster I'm curious how it will behave in that case. We maybe need to ask the team to test on bigger cluster as well.
> 
> @Pegonzal @aaSharma14 @nizamial09

yeah, for now let's put 60 to the affected calls and run some tests with it. and based on the results we can increase if needed!

File: src/pybind/mgr/mgr_module.py [hunk 1]
```diff
@@ -2351,7 +2351,8 @@ def is_authorized(self, arguments: Dict[str, str]) -> bool:
 
     @API.expose
     def send_rgwadmin_command(self, args: List[str],
-                              stdout_as_json: bool = True) -> Tuple[int, Union[str, dict], str]:
+                              stdout_as_json: bool = True,
+                              timeout: Optional[int] = 20) -> Tuple[int, Union[str, dict], str]:
```
Comment (nizamial09) [inline src/pybind/mgr/mgr_module.py:2355]: do we need to increase the default? whatever commands are affected that can be put to 30 for now?

File: src/pybind/mgr/dashboard/services/rgw_client.py [hunk 2]
```diff
@@ -1280,7 +1280,7 @@ def edit_zonegroup(self, realm_name: str, zonegroup_name: str, new_zonegroup_nam
     def update_period(self):
         rgw_update_period_cmd = ['period', 'update', '--commit']
         try:
-            exit_code, _, err = mgr.send_rgwadmin_command(rgw_update_period_cmd)
+            exit_code, _, err = mgr.send_rgwadmin_command(rgw_update_period_cmd, True, 60)
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/services/rgw_client.py:1283]:
```suggestion
            exit_code, _, err = mgr.send_rgwadmin_command(rgw_update_period_cmd, timeout=30)
```

**Comment by aaSharma14** [comment]
jenkins test make check

File: src/pybind/mgr/dashboard/services/rgw_client.py [hunk 3]
```diff
@@ -1280,7 +1280,7 @@ def edit_zonegroup(self, realm_name: str, zonegroup_name: str, new_zonegroup_nam
     def update_period(self):
         rgw_update_period_cmd = ['period', 'update', '--commit']
         try:
-            exit_code, _, err = mgr.send_rgwadmin_command(rgw_update_period_cmd)
+            exit_code, _, err = mgr.send_rgwadmin_command(rgw_update_period_cmd, True, timeout=30)
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/services/rgw_client.py:1283]: why do we set some unknown variable to `True`. Its best to change whatever we need.
```suggestion
            exit_code, _, err = mgr.send_rgwadmin_command(rgw_update_period_cmd, timeout=30)
```

**Comment by nizamial09** [comment]
@aaSharma14 this doesn't have a tracker as well. please attach it to the commit message

**Pull request closed without merging**